src/main.rs

```diff
@@ -1,125 +1,43 @@
 mod tagsdb;
-use tagsdb::{Entry, TagsDb};
-
-use itertools::Itertools;
-use once_cell::sync::Lazy;
-use percent_encoding::{percent_encode, NON_ALPHANUMERIC};
-use regex::Regex;
-use std::{env, sync::Arc};
-use teloxide::{prelude::*, types::ParseMode, utils::html};
-
-macro_rules! help_regex_s {
-    () => {
-        r":he?l?p?\s+([^\s]+)"
-    };
-}
-
-static HELP_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(help_regex_s!()).expect("regex compiles"));
-
-static DELETE_REGEX: Lazy<Regex> = Lazy::new(|| {
-    Regex::new(concat!(r"^[\s\n]*(?:", help_regex_s!(), r"[\s\n]*)+$")).expect("regex compiles")
-});
-
-static VIM_URL_BASE: &str = "https://vimhelp.org";
-static NEOVIM_URL_BASE: &str = "https://neovim.io/doc/user";
-
-#[derive(Clone, Copy, Debug)]
-enum Flavor {
-    Vim,
-    NeoVim,
-}
-
-use Flavor::*;
-
-fn format_vim_url(entry: &Entry) -> String {
-    format!(
-        "{}/{}.txt.html#{}",
-        VIM_URL_BASE,
-        entry.filename,
-        percent_encode(entry.topic.as_bytes(), NON_ALPHANUMERIC)
-    )
-}
-
-fn format_neovim_url(entry: &Entry) -> String {
-    format!(
-        "{}/{}.html#{}",
-        NEOVIM_URL_BASE,
-        entry.filename,
-        percent_encode(entry.topic.as_bytes(), NON_ALPHANUMERIC)
-    )
-}
-
-fn format_message(links: impl IntoIterator<Item = (Entry, Flavor)>) -> String {
-    links
-        .into_iter()
-        .map(|(entry, flavor)| {
-            let url = match flavor {
-                Vim => format_vim_url(&entry),
-                NeoVim => format_neovim_url(&entry),
-            };
-            format!(
-                "Found help for {} in {:?} docs:\n{}",
-                html::code_inline(&entry.topic),
-                flavor,
-                url
-            )
-        })
-        .join("\n\n")
+mod tagsearch;
+mod utils;
+
+use std::env;
+use tagsearch::TagSearch;
+use teloxide::{prelude::*, types::ParseMode};
+
+async fn handle_message(message: UpdateWithCx<Message>, tagsearch: TagSearch) {
+    if let Some(text) = message.update.text() {
+        if let Some(answer) = tagsearch.find_by_text(text) {
+            let send_message = message.answer(answer).parse_mode(ParseMode::HTML);
+            if let Some(reply) = message.update.reply_to_message() {
+                let sent_ans = send_message.reply_to_message_id(reply.id);
+                if let Ok(_) = sent_ans.send().await {
+                    message.delete_message().send().await.ok();
```

Comment:
With this change bot won't delete message if it's _not_ a reply — I think, it's incorrect behaviour.